db/repair.cc

```diff
@@ -163,31 +164,40 @@ class Repairer {
     return status;
   }
 
-  ~Repairer() {
-    if (db_lock_ != nullptr) {
-      env_->UnlockFile(db_lock_);
+  Status Close() {
+    Status s = Status::OK();
+    if (!closed_) {
+      if (db_lock_ != nullptr) {
+        s = env_->UnlockFile(db_lock_);
```

Comment:
What happens if the Unlock fails?  The file will be marked as Closed but the db_lock_ is still hanging out?

Reply:
Fixed to set to nullptr since `UnlockFile()` is not retryable and deletes the pointer passed to it even in the failure case.